src/seq.rs

```diff
@@ -475,49 +480,97 @@ impl PortSubscribe {
 
 }
 
+#[derive(Clone)]
+enum EventExtraData<'a> {
+    None,
+    Borrowed(&'a [u8]),
+    Owned(Vec<u8>)
+}
+
+impl<'a> EventExtraData<'a> {
+    fn into_owned<'b>(self) -> EventExtraData<'b> {
+        match self {
+            EventExtraData::None => EventExtraData::None,
+            EventExtraData::Borrowed(buf) => EventExtraData::Owned(buf.to_vec()),
+            EventExtraData::Owned(vec) => EventExtraData::Owned(vec)
+        }
+    }
+}
+
 /// [snd_seq_event_t](http://www.alsa-project.org/alsa-doc/alsa-lib/structsnd__seq__event__t.html) wrapper
 ///
 /// Fields of the event is not directly exposed. Instead call `Event::new` to set data (which can be, e g, an EvNote).
 /// Use `get_type` and `get_data` to retreive data.
-pub struct Event(alsa::snd_seq_event_t, EventType, Option<Vec<u8>>);
+///
+/// The lifetime parameter refers to the lifetime of an associated external buffer that might be used for
+/// variable-length messages (e.g. SysEx).
+pub struct Event<'a>(alsa::snd_seq_event_t, EventType, EventExtraData<'a>);
```

Comment:
That would be equivalent, I think, but in my opinion the extra struct makes the intent clearer to the reader of the code. I'm fine with either way, since this is an implementation detail (the `EventExtraData` struct should never be exposed as part of the public API).